pyuvdata/tests/test_miriad.py

```diff
@@ -563,9 +559,49 @@ def test_readWriteReadMiriad():
     nt.assert_equal(nschan, nfreqs)
     nt.assert_equal(ischan, 1)
 
+    # check partial IO selections
+    full = UVData()
+    full.read_miriad(testfile)
+    uv_in = UVData()
+
+    # test only specified bls were read, and that flipped antpair is loaded too
+    uv_in.read_miriad(testfile, antpairs=[(0, 0), (0, 1), (4, 2)])
+    nt.assert_equal(uv_in.get_antpairs(), [(0, 0), (0, 1), (2, 4)])
+
```

Comment:
Can you also test that the object overall is equal if you read in the whole file and then do a matching select for all these partial read tests? That's a much more comprehensive test that may catch edge cases. (You may have to have some special handling for the history parameter, you can see how we do that for uvfits if you want.)

Reply:
sure thing